utils/build.ps1

```diff
@@ -402,96 +407,98 @@ function Invoke-VsDevShell($Arch) {
+      if ($SHA256.Hash -ne $WiXHash) {
+        throw "WiX SHA256 mismatch ($($SHA256.Hash) vs $WiXHash)"
+      }
 
-  foreach ($Arch in $WinSDKArchs) {
-    $WinSDKArchPackageName = "$WinSDKBasePackageName.$($Arch.ShortName)"
-    Invoke-Program nuget install $WinSDKArchPackageName -Version $WinSDKVersion -OutputDirectory $NugetRoot
-    Copy-Directory "$NugetRoot\$WinSDKArchPackageName.$WinSDKVersion\c\*" "$CustomWinSDKRoot\lib\$WinSDKVersionRevisionZero"
+      New-Item -ItemType Directory -ErrorAction Ignore $BinaryCache\WiX-$WiXVersion | Out-Null
+      Write-Output "Extracting WiX ..."
+      Expand-Archive -Path $BinaryCache\WiX-$WiXVersion.zip -Destination $BinaryCache\WiX-$WiXVersion -Force
+    }
   }
-}
 
-function Ensure-SwiftToolchain($Arch) {
-  if (-not (Test-Path $BinaryCache\wix-4.0.1.zip)) {
-    Write-Output "WiX not found. Downloading from nuget.org..."
-    Invoke-Program curl.exe -sL https://www.nuget.org/api/v2/package/wix/4.0.1 --output $BinaryCache\wix-4.0.1.zip --create-dirs
-  }
+  if (-not (Test-Path $BinaryCache\$PinnedToolchain.exe)) {
+    Write-Output "Swift toolchain not found. Downloading from swift.org..."
+    if ($ToBatch) {
+      Write-Output "curl.exe -sL $PinnedBuild -o $BinaryCache\$PinnedToolchain.exe"
+    } else {
+      $WebClient.DownloadFile($PinnedBuild, $BinaryCache\$PinnedToolchain.exe)
+      $SHA256 = Get-FIleHash -Path "$BinaryCache\$PinnedToolchain.exe" -Algorithm SHA256
+      if ($SHA256.Hash -ne $PinnedSHA256) {
+        throw "$PinnedToolchain SHA256 mismatch ($($SHA256.Hash) vs $PinnedSHA256)"
+      }
 
-  if (-not $ToBatch) {
-    $SHA256 = Get-FileHash -Path "$BinaryCache\wix-4.0.1.zip" -Algorithm SHA256
-    if ($SHA256.Hash -ne "756AD3115F0CE808313266F4E401C0F520D319211DE0B9D8D7E7697020E0C461") {
-      throw "WiX SHA256 mismatch ($($SHA256.Hash) vs 756AD3115F0CE808313266F4E401C0F520D319211DE0B9D8D7E7697020E0C461)"
+      New-Item -ItemType Directory -ErrorAction Ignore $BinaryCache\toolchains | Out-Null
+      # The new runtime MSI is built to expand files into the immediate directory. So, setup the installation location.
+      New-Item -ItemType Directory -ErrorAction Ignore $BinaryCache\toolchains\$PinnedToolchain\LocalApp\Programs\Swift\Runtimes\0.0.0\usr\bin | Out-Null
     }
-  }
 
-  New-Item -ItemType Directory -ErrorAction Ignore $BinaryCache\wix-4.0.1 | Out-Null
-  Write-Output "Extracting WiX..."
-  Expand-Archive -Path $BinaryCache\wix-4.0.1.zip -Destination $BinaryCache\wix-4.0.1 -Force
-
-  if (-not (Test-Path "$BinaryCache\${PinnedToolchain}.exe")) {
-    Write-Output "Swift toolchain not found. Downloading from swift.org..."
-    (New-Object Net.WebClient).DownloadFile($PinnedBuild, "$BinaryCache\${PinnedToolchain}.exe")
-    # Invoke-Program curl.exe -sL $PinnedBuild --output $BinaryCache\${PinnedToolchain}.exe --create-dirs
-  }
+    Write-Output "Extracting $PinnedToolchain ..."
```

Comment:
Is there ToBatch case code to add for the toolchain part?

Reply:
The `Invoke-Program` will implicitly handle the `ToBatch`.